backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/starrating/BpkStarRating.kt

```diff
@@ -0,0 +1,147 @@
+
+enum class BpkStarRatingSize {
+  Large,
+  Small,
+}
+
+@Composable
+fun BpkStarRating(
+  rating: Float,
+  contentDescription: ContentDescriptionScope.(Float, Int) -> String,
+  modifier: Modifier = Modifier,
+  rounding: RoundingType = RoundingType.Down,
+  size: BpkStarRatingSize = BpkStarRatingSize.Small,
+) {
+  BpkStarRating(
+    maxRating = 5,
+    rounding = rounding,
+    iconSize = when (size) {
+      BpkStarRatingSize.Large -> BpkIconSize.Large
+      BpkStarRatingSize.Small -> BpkIconSize.Small
+    },
+    modifier = modifier,
+    rating = rating,
+    contentDescription = contentDescription,
+  )
+}
+
+@Composable
+fun BpkHotelRating(
+  rating: Int,
+  contentDescription: ContentDescriptionScope.(Float, Int) -> String,
+  modifier: Modifier = Modifier,
+  size: BpkStarRatingSize = BpkStarRatingSize.Small,
+) {
+  BpkStarRating(
+    maxRating = rating,
+    rounding = RoundingType.Down,
+    iconSize = when (size) {
+      BpkStarRatingSize.Large -> BpkIconSize.Large
+      BpkStarRatingSize.Small -> BpkIconSize.Small
+    },
+    modifier = modifier,
+    rating = rating.toFloat(),
+    contentDescription = contentDescription,
+  )
+}
+
+@Composable
+private fun BpkStarRating(
+  rating: Float,
+  maxRating: Int,
+  rounding: RoundingType,
+  iconSize: BpkIconSize,
+  contentDescription: ContentDescriptionScope.(Float, Int) -> String,
+  modifier: Modifier = Modifier,
+) {
+  val coercedRating = rating.coerceIn(0f, maxRating.toFloat())
+  val roundedRating = when (rounding) {
+    RoundingType.Down -> floor(coercedRating * 2) / 2
+    RoundingType.Up -> ceil(coercedRating * 2) / 2
+    RoundingType.Nearest -> round(coercedRating * 2) / 2
+  }
+  val scope = rememberContentDescriptionScope()
+  Row(modifier = modifier.semantics { this.contentDescription = scope.contentDescription(roundedRating, maxRating) }) {
+    for (item in 0 until maxRating) {
+      val value = (roundedRating - item).coerceIn(0f, 1f)
+      when {
+        (value >= 0.0f && value < 0.5f) -> BpkStar(icon = StarType.Empty, iconSize = iconSize)
+        (value >= 0.5f && value < 1.0f) -> BpkStar(icon = StarType.Half, iconSize = iconSize)
+        else -> BpkStar(icon = StarType.Full, iconSize = iconSize)
+      }
+    }
+  }
+}
+
+@Composable
+private fun BpkStar(icon: StarType, iconSize: BpkIconSize) {
+  when (icon) {
+    StarType.Empty -> BpkIcon(
+      icon = BpkIcon.StarOutline,
+      contentDescription = null,
+      size = iconSize,
+      tint = BpkTheme.colors.textDisabled,
+    )
+    StarType.Half -> BpkIcon(
+      icon = BpkIcon.StarHalf,
+      contentDescription = null,
+      size = iconSize,
+      tint = BpkTheme.colors.statusWarningSpot,
+    )
+    StarType.Full -> BpkIcon(
+      icon = BpkIcon.Star,
+      contentDescription = null,
+      size = iconSize,
+      tint = BpkTheme.colors.statusWarningSpot,
+    )
+  }
+}
+
+private enum class StarType {
+  Empty,
+  Half,
+  Full,
+}
+
+enum class RoundingType {
```

Comment:
Can we have this enum in the beginning of the file, in the same way, we do for Size? Also, let's keep the naming consistent: BpkRatingRounding